hypershift-operator/controllers/hostedcluster/ignitionserver/ignitionserver.go

```diff
@@ -475,33 +475,13 @@ func reconcileDeployment(deployment *appsv1.Deployment,
 		deployment.Annotations = map[string]string{}
 	}
 
-	// Before this change we did
-	// 		Selector: &metav1.LabelSelector{
-	//			MatchLabels: ignitionServerLabels,
-	//		},
-	//		Template: corev1.PodTemplateSpec{
-	//			ObjectMeta: metav1.ObjectMeta{
-	//				Labels: ignitionServerLabels,
-	//			}
-	// As a consequence of using the same memory address for both MatchLabels and Labels, when setColocation set the colocationLabelKey in additionalLabels
-	// it got also silently included in MatchLabels. This made any additional additionalLabel to break reconciliation because MatchLabels is an immutable field.
-	// So now we leave Selector.MatchLabels if it has something already and use a different var from .Labels so the former is not impacted by additionalLabels changes.
-	selectorLabels := ignitionServerLabels
-	if deployment.Spec.Selector != nil && deployment.Spec.Selector.MatchLabels != nil {
-		selectorLabels = deployment.Spec.Selector.MatchLabels
-	}
-
 	deployment.Spec = appsv1.DeploymentSpec{
 		Selector: &metav1.LabelSelector{
-			MatchLabels: selectorLabels,
+			MatchLabels: ignitionServerLabels,
```

Comment:
This reverts back to the previous behavior with the unintended side effect of modifying both sets by using the same memory location. Can we instead use:
```
config.CopyStringMap(ignitionServerLabels)
```
in both locations?

Reply:
That should potentially work seeing as it was populating the pod template accurately, and then they'd have separate memory addresses. I'll give that a shot--thanks!

Reply:
actually, after thinking about this further, I think we'd still fall into the same issue since when it does the `config.CopyStringMap(ignitionServerLabels)`, it'll only populate the default labels, and the additional labels will still only be applied to the pod template via the `deploymentConfig.ApplyTo(deployment)` flow.

Reply:
That was the original intent (that they only be applied to the pod template), the selector was meant to remain as simple as possible. We got into the issue because we were using the same memory address.